### ISSUE TITLE
Artifacts / Attachments Feature

### ISSUE THREAD

**Issue by carlosbaraza** [gitlabhq/gitlab-ci#597]
# Problem
Currently in my company we are maintaining a Ruby application. We use PhantomJS to run a Cucumber/RSpec e2e test suite. When a test fails, we render a screenshot of the page and we save the html code at the failure point.

It would be really awesome to be able to embed files in the build results page, to ease the access to these kind of Artifacts / Attachments.

# Possible implementation
A possible way I though about is to create a folder for each build id. All the files in the folder for a specific build would be published and pointed from the build results page.

# Notes
I could implement the feature if validated by the GitLab team. What do you think?

Thank you for this awesome tool.
Carlos.

**Comment by ayufan** [comment]
Maybe in general possibility to push artifacts to CI? It could be nicely done in the upcoming gitlab-ci.yml :)

**Comment by vsizov** [comment]
Yeah, it would be cool. I think we have this planned but not in coming releases.

**Comment by dosire** [comment]
I would love this feature http://feedback.gitlab.com/forums/176466-general/suggestions/4522830-allow-access-to-build-artifacts-of-gitlab-ci

I think we should use https://github.com/travis-ci/artifacts

**Comment by ayufan** [comment]
@dosire this is really nice and pretty easy to use: https://github.com/travis-ci/dpl

**Comment by dosire** [comment]
@ayufan yes, that too indeed!

**Comment by dosire** [comment]
@ayufan or start with https://github.com/travis-ci/dpl
